react-native-aztec/ios/RNTAztecView/RCTAztecView.swift

```diff
@@ -69,23 +97,33 @@ class RCTAztecView: Aztec.TextView {
 
     func commonInit() {
         delegate = self
+        addPlaceholder()
+    }
+
+    func addPlaceholder() {
         addSubview(placeholderLabel)
-        placeholderLabel.textAlignment = .natural
-        placeholderLabel.translatesAutoresizingMaskIntoConstraints = false
-        placeholderLabel.font = font
         NSLayoutConstraint.activate([
-            placeholderLabel.leadingAnchor.constraint(equalTo: leadingAnchor, constant: contentInset.left + textContainerInset.left + textContainer.lineFragmentPadding),
+            placeholderHorizontalConstraint,
             placeholderLabel.topAnchor.constraint(equalTo: topAnchor, constant: contentInset.top + textContainerInset.top)
-            ])
+        ])
     }
 
     // MARK - View Height: Match to content height
-    
+
     override func layoutSubviews() {
         super.layoutSubviews()
+        fixLabelPositionForRTLLayout()
```

Comment:
This call here has me a bit confused because of the asymmetry between how we handle LTR and RTL: for LTR we have a fixed constraint setup at view-initialization time (well almost because it's lazy, but I digress), whereas for RTL we can update it dynamically whenever `layoutSubviews` is called.

In my mind if we need a dynamic solution, we need both updates here, whereas if a static solution is fine, we could move this code to the lazy initializer of `placeholderHorizontalConstraint` and make it something like:

```swift
    private lazy var placeholderHorizontalConstraint: NSLayoutConstraint = {
        return placeholderPreferedHorizontalAnchor.constraint(
            equalTo: leftAnchor,
            constant: isRTLLayout ? rightTextInset : leftTextInset
        )
    }()
```

Does that make sense?  Let me know if there's more to it.

Reply:
This asymmetry is actually needed because of the horizontal flipping of the scroll views on RTL layout.
When we try to add a view on a horizontally flipped view, what happens is that the left and right right sides are also flipped logically, but RN flips back the child views so they don't look reversed.

When we try to constraint the placeholder's right anchor to the parent's right anchor (as we would normally do) this happens:

<img width="526" alt="screen shot 2019-02-15 at 6 20 45 pm" src="https://user-images.githubusercontent.com/9772967/52873391-2ede2a80-314f-11e9-8621-5d57b0345131.png">

Logically it's correctly anchoring it to the right, but it's visual positioned at the left. You can see the long constraint across the Header, that it's set to 0, and breaks constraints ¯\_(ツ)_/¯

That's why we set the right anchor of the left anchor of the parent, that will give us this:

<img width="512" alt="screen shot 2019-02-15 at 7 07 24 pm" src="https://user-images.githubusercontent.com/9772967/52875587-f5a8b900-3154-11e9-9314-9d2b724edb83.png">

This is not perfect yet, but now the horizontal constraint will respond properly, so we can work with it.
What we have to do is to move the label all the way to the other side of the screen. That's when the dynamic value comes to play.
We set the constant of the constraint to be the width of the parent, and if the parent changes, this has to update accordingly. That's why it's in `layoutSubviews`.

In the other hand, for LTR, a static value will be enough, since constraining left to left anchors won't need modifications if the layout changes.

I hope this makes clearer.

If we just want consistency, we can always update "dynamically" on both RTL and LTR layouts, but LTR is not necessary, and I think it is good to remark this difference.

I see that my code comments weren't clear enough 😅